flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -309,6 +312,25 @@ public TaskExecutor(
 		return CompletableFuture.completedFuture(shuffleEnvironment.getPartitionsOccupyingLocalResources().isEmpty());
 	}
 
+	@Override
+	public CompletableFuture<Collection<LogInfo>> requestLogList(Time timeout) {
+		final String logDir = taskManagerConfiguration.getTaskManagerLogDir();
+		if (logDir != null) {
+			final File[] logFiles = new File(logDir).listFiles();
```

Comment:
Unfortunately this is a blocking IO operation in the main thread which is discouraged. But the code was already broken before as we are uploading the files in the main thread. I'll have to think about whether this needs to be addressed immediately.

Reply:
We can continue for now but FLINK-16710 should be fixed next.